### PR TITLE
chore(): Refactor forms to use zod (part I)

### PR DESCRIPTION
**Pull request by matteolc**
### Ticket:

- [23703](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/23703)

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [ ] **New feature**
- [ ] **Dev change**
- [ ] **Additional tests**
- [ ] **Documentation**
- [ ] **Other**

### Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This change might impact the developer experience of others and should be communicated

### Screenshots:

N/A

### CODE REVIEW

File: apps/messaging/app/[locale]/admin/users/imports/[importId]/page.tsx [hunk 1]
```diff
@@ -19,8 +20,7 @@ export default async (props: {
     getTranslations("UsersImport"),
     getTranslations("Commons"),
   ])
-  const { organization } =
-    await AuthenticationFactory.getInstance().getPublicServant()
+  const organization = await getOrganisationOrThrow()
```
Comment (Thurfjell) [inline apps/messaging/app/[locale]/admin/users/imports/[importId]/page.tsx:23]: If this throws, it wont be caught by any boundaries and crash the page. Try/catch with let?

File: apps/messaging/app/utils/auth.ts [hunk 2]
```diff
@@ -1,15 +1,65 @@
...
+  const user = await authentication.getUser()
+  return {
+    user,
+    isPublicServant: await authentication.isPublicServant(),
+    isInactivePublicServant: await authentication.isInactivePublicServant(),
+  }
+}
+
+const getPublicServantOrRedirect = async ({
+  redirectUserTo,
```
Comment (Thurfjell) [inline apps/messaging/app/utils/auth.ts:16]: or maybe redirectFallback? a ps is a user too, might be confusing in a month?

File: apps/messaging/components/send-message/Recipients.tsx [hunk 3]
```diff
@@ -47,14 +47,15 @@ export default () => {
     }
   }, [message, onStep, userIds])
 
+  // biome-ignore lint/correctness/useExhaustiveDependencies: <explanation>
```
Comment (Thurfjell) [inline apps/messaging/components/send-message/Recipients.tsx:50]: 😞

File: apps/messaging/app/[locale]/admin/users/imports/[importId]/page.tsx (hunk 1 above)
Comment (matteolc) [reply apps/messaging/app/[locale]/admin/users/imports/[importId]/page.tsx:23]: My aim here was to refactor the common function, not change the functionality which should be as it was before. We can tackle this server errors in a follow-up PR once we've got to a consensus on how to handle them?

File: apps/messaging/components/providers/types.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  name: z.string().min(1, { message: "Provider name is required" }),
+  host: z.string().min(1, { message: "Host is required" }),
+  port: z.number().min(1, { message: "Port is required" }),
+  username: z.string().min(1, { message: "Username is required" }),
+  password: z.string().min(1, { message: "Password is required" }),
+  fromAddress: z.string().email({ message: "From address is required" }),
+  throttle: z.number().optional(),
+  ssl: z.boolean().default(false),
+  isPrimary: z.boolean().default(false),
+})
```
[the remark this answers is not in the document]
Comment (matteolc) [reply apps/messaging/components/providers/types.ts:14]: Yes

File: apps/messaging/components/send-message/Recipients.tsx (hunk 3 above)
Comment (matteolc) [reply apps/messaging/components/send-message/Recipients.tsx:50]: I know. I don't like to ignore exhaustive deps. In this case query params where causing an infinite re-render.

File: apps/messaging/app/[locale]/admin/users/imports/[importId]/page.tsx (hunk 1 above)
Comment (Thurfjell) [reply apps/messaging/app/[locale]/admin/users/imports/[importId]/page.tsx:23]: Perfect!

Comment (matteolc) [reply apps/messaging/app/[locale]/admin/users/imports/[importId]/page.tsx:23]: There was an extra throw after I have removed:
https://github.com/ogcio/life-events/commit/631f4b4d8181208302c1f95f29192e4084d4c47c

File: apps/messaging/app/utils/auth.ts (hunk 2 above)
Comment (matteolc) [reply apps/messaging/app/utils/auth.ts:16]: If you want a user, you can use the `getUser` function

Comment (Thurfjell) [reply apps/messaging/app/utils/auth.ts:16]: Yes, but you also have an optional redirect for public servant. It's purely from my own perspective, I'd be confused in 3 months time, and would have to read the definition.

A public servant is also a user of the system.

Quite nitty though so let's move on.

**Review by Thurfjell — APPROVED** [review]

File: apps/messaging/components/providers/types.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (matteolc) [reply apps/messaging/components/providers/types.ts]: https://github.com/ogcio/life-events/commit/d306dabd133878bb0d62980daaab926761fedfb2

**Comment by Thurfjell** [comment]
Just to be crystal clear. 

We're talking about client side forms with some javascript event handler function, and not the next server action types of form?

**Review by Thurfjell — APPROVED** [review]

**Pull request merged**